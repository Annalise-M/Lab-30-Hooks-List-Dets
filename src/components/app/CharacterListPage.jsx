/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useCharacters } from '../../hooks/characters';
import CharacterItem from './CharacterItem';



const CharacterListPage = () => {
  const { characters } = useCharacters();

  const characterElements = characters.map(character => (
    <li key={character._id}>
      <CharacterItem {...character} />
    </li>
  ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterListPage.propTypes = {

};

export default CharacterListPage;
