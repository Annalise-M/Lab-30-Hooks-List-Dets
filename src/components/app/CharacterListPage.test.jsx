import React from 'react';
import { render, cleanup, waitFor, screen } from '@testing-library/react';
import CharacterListPage from './CharacterListPage';

describe('CharacterListPage Container', () => {
  afterEach(() => cleanup());
  it('displays HeyArnold character data', () => {
    const { asFragment } = render(<CharacterListPage />);
    expect(asFragment()).toMatchSnapshot();
  });

});

describe('CharacterListPage Component', () => {
  it('displays a list of characters', async() => {
    render(<CharacterListPage />);

    const characterList = await screen.findByTestId('characters');

    return waitFor(() => {
      expect(characterList).not.toBeEmptyDOMElement();
    });
  });
});

