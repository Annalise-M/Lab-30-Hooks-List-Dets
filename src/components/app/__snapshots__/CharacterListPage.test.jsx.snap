// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListPage Container displays HeyArnold character data 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  />
</DocumentFragment>
`;
